components/client/src/main/java/com/hotels/styx/client/StyxClientError.java

```diff
@@ -0,0 +1,31 @@
+/**
+ * Copyright (C) 2013-2017 Expedia Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hotels.styx.client;
+
+/**
+ * Signifies an unexpected error from StyxClient.
+ *
+ * A root cause for the error is stored in the `cause` field.
+ * This could be thrown for example if StyxClient runs out of
+ * direct memory.
+ *
+ */
+public class StyxClientError extends RuntimeException {
```

Comment:
In Java, an Error and an Exception are not the same thing. We should ensure that our throwable class names match their parent types.